kaggle/src/connector.js

```diff
@@ -0,0 +1,291 @@
+
+function getAuthType() {
+  return {
+    "type": "USER_PASS"
+  };
+}
+
+function getConfig(request) {
+  var config = {
+    configParams: [
+        {
+            'type': 'INFO',
+            'name': 'generalInfo',
+            'text': 'Enter the following information for the desired Kaggle dataset. The kaggle URL for datasets gives the ownerSlug and datasetSlug. For eg: https://www.kaggle.com/{ownerSlug}/{datasetSlug} '
+        },
+        {
+            'type': 'TEXTINPUT',
+            'name': 'ownerSlug',
+            'displayName': 'Enter owner slug',
+            'placeholder': connector.ownerSlug
+        },
+        {
+            'type': 'TEXTINPUT',
+            'name': 'datasetSlug',
+            'displayName': 'Enter dataset slug',
+            'placeholder': connector.datasetSlug
+        },
+        {
+            'type': 'TEXTINPUT',
+            'name': 'fileName',
+            'displayName': 'Enter filename',
+            'placeholder': connector.fileName
+        }
+    ] 
+  };
+  return config;
+}
+
+function getSchema(request) {
+  request = validateConfig(request);
+  var rawData = getFileData(request.configParams);
+  var kaggleSchema = buildSchema(rawData);
+  return { schema: kaggleSchema };
+}
+
+function validateConfig(request) {
+  request.configParams = request.configParams || {}; 
+  var config = request.configParams;
+  config.ownerSlug = config.ownerSlug || connector.ownerSlug;
+  config.datasetSlug = config.datasetSlug || connector.datasetSlug;
+  config.fileName = config.fileName || connector.fileName;
+  return request;
+}
+
+function getData(request) {
+  request = validateConfig(request);
+  var rawData = getFileData(request.configParams);
+  var kaggleSchema = buildSchema(rawData);
+  
+  var requestedSchema = request.fields.map(function (field) {
+    for (var i = 0; i < kaggleSchema.length; i++) {
+      if (kaggleSchema[i].name == field.name) {
+        return kaggleSchema[i];
+      }
+    }
+  });
+  
+  var requestedData = processData(rawData, requestedSchema);
+  
+  return {
+    schema: requestedSchema,
+    rows: requestedData
+  };
+}
+
+function buildSchema(data) {
+  var columnNames = data[0];
+  var content = data[1];
+  
+  var schema = [];
+  for (var i = 0; i < columnNames.length; i++) {
+    var fieldSchema = mapColumn(i, columnNames[i], content[i]);
+    schema.push(fieldSchema);
+  }
+  return schema;
+}
+
+function mapColumn(index, columnName, content) {
+  var field = {};
+  field.name = 'c' + index;
+  field.label = columnName;
+  if (isNaN(content)) {
+    field.dataType = 'STRING';
+    field.semantics = {};
+    field.semantics.conceptType = 'DIMENSION';
+  } else {
+    field.dataType = 'NUMBER';
+    field.semantics = {};
+    field.semantics.conceptType = 'METRIC';
+    field.semantics.isReaggregatable = true
+  };
+  return field;
+};
+
+function processData(data, fields) {
+  var header = data[0];
+  var dataIndexes = fields.map( function(field){
```

Comment:
Nit: please remove the extra whitespace before `function`

Reply:
Actually, just run `npm run prettier` from the top level, it'll take care of all of the style things.